# 2ª PARTE

In [1]:
import pandas as pd

In [2]:
vuelos=pd.read_csv('DelayedFlights.csv')

In [3]:
#Reducción a 200000 registros (filas) para trabajar en todos los ejercicios
vuelos=vuelos.sample(200000)

## Ejercicio 1

In [4]:
## Visualización de todas las variables existentes en el DataFrame para la elección de las que tengan información 
## relevante para su análisis estadístico.
vuelos.columns

Index(['Unnamed: 0', 'Year', 'Month', 'DayofMonth', 'DayOfWeek', 'DepTime',
       'CRSDepTime', 'ArrTime', 'CRSArrTime', 'UniqueCarrier', 'FlightNum',
       'TailNum', 'ActualElapsedTime', 'CRSElapsedTime', 'AirTime', 'ArrDelay',
       'DepDelay', 'Origin', 'Dest', 'Distance', 'TaxiIn', 'TaxiOut',
       'Cancelled', 'CancellationCode', 'Diverted', 'CarrierDelay',
       'WeatherDelay', 'NASDelay', 'SecurityDelay', 'LateAircraftDelay'],
      dtype='object')

Como el objetivo del nuevo DataFrame es su análisis estadístico sobre los tiempos de los vuelos, he eliminado las variables (o columnas) que carecen de tal fin, tales como:

'Unnamed'; 

'Year', ya que todos los registros corresponden al año 2008; 

'FlightNUm' y 'TailNum', ambas solo contienen información de interés al ámbito de control interno para las compañías aéreas;

'TaxiIn' y 'TaxiOout', ya están dentro del tiempo de vuelo;

'Cancelled', 'CancellationCode' y 'Diverted', el dataframe no registra datos;

'CarrierDelay', 'WeatherDelay', NASDelay', ' 'SecurityDelay' y 'LateAircraftDelay', aquí no analizamos la responsabilidad por el retraso del vuelo, sino los tiempos que se han empleado (además, muchos de sus valores carecen de registro).

### Mi tabla quedaría de la siguiente manera:

In [5]:
vuelos_colum_relev=vuelos[['UniqueCarrier', 'Month', 'DayofMonth', 'DayOfWeek', 'CRSDepTime', 'DepTime',
       'CRSArrTime', 'ArrTime', 'CRSElapsedTime', 'ActualElapsedTime', 'AirTime', 'DepDelay', 
        'ArrDelay', 'Origin', 'Dest', 'Distance']]

vuelos_colum_relev

UniqueCarrier  Month  DayofMonth  DayOfWeek  CRSDepTime  DepTime  \
100581             EV      1          28          1        1007   1015.0   
1403335            AS      8           5          2        1845   1852.0   
253588             OO      2          29          5        1710   1918.0   
220528             XE      2          10          7        1351   1401.0   
1508200            AA      9           9          2        1740   1749.0   
...               ...    ...         ...        ...         ...      ...   
437101             OO      3           5          3        1015   1101.0   
1286229            WN      8          22          5        2015   2027.0   
1103904            WN      7          24          4        2115   2148.0   
1614296            AS     10          10          5        1130   1202.0   
191828             WN      2           8          5        1355   1411.0   

         CRSArrTime  ArrTime  CRSElapsedTime  ActualElapsedTime  AirTime  \
100581         1148   1142.0           161.0              147.0    119.0   
1403335        2145   2147.0           180.0              175.0    162.0   
253588         1851   2048.0           101.0               90.0     73.0   
220528         1511   1512.0            80.0               71.0     49.0   
1508200        1905   1854.0           145.0              125.0    103.0   
...             ...      ...             ...                ...      ...   
437101         1221   1254.0           126.0              113.0     97.0   
1286229        2230   2229.0           135.0              122.0    109.0   
1103904        2245   2316.0            90.0               88.0     73.0   
1614296        1554   1627.0           204.0              205.0    175.0   
191828         1520   1546.0            85.0               95.0     55.0   

         DepDelay  ArrDelay Origin Dest  Distance  
100581        8.0      -6.0    ATL  AUS       813  
1403335       7.0       2.0    TUS  SEA      1216  
253588      128.0     117.0    ORD  FSD       462  
220528       10.0       1.0    LAX  SJC       308  
1508200       9.0     -11.0    ATL  DFW       732  
...           ...       ...    ...  ...       ...  
437101       46.0      33.0    DEN  MSO       679  
1286229      12.0      -1.0    TPA  CMH       829  
1103904      33.0      31.0    BNA  MCI       491  
1614296      32.0      33.0    ANC  SEA      1449  
191828       16.0      26.0    LAX  SFO       337  

[200000 rows x 16 columns]

## Ejercicio 2

In [6]:
# a) Resumen estadístico y análisis de datos relevantes (no habituales o evidentes)
vuelos_colum_relev[['CRSDepTime', 'ActualElapsedTime',
                    'AirTime', 'DepDelay', 'ArrDelay', 'Distance']].describe().round(2)

CRSDepTime  ActualElapsedTime    AirTime   DepDelay   ArrDelay  \
count   200000.00          199103.00  199103.00  200000.00  199103.00   
mean      1468.59             133.56     108.49      43.18      42.24   
std        424.29              72.03      68.59      53.32      56.76   
min          1.00              15.00       1.00       6.00     -66.00   
25%       1135.00              81.00      58.00      12.00       9.00   
50%       1510.00             116.00      90.00      24.00      24.00   
75%       1819.00             165.00     138.00      53.00      56.00   
max       2359.00             681.00     645.00    2467.00    2453.00   

        Distance  
count  200000.00  
mean      766.81  
std       573.23  
min        31.00  
25%       341.00  
50%       607.00  
75%       998.00  
max      4962.00

Como puede observarse, la mitad de los vuelos se efectúan pasado poco más el mediodía (15:10), y solo tres horas más tarde (18:19) ya se han realizado el 75%.

Por otra parte, la media del transcurso de los vuelos es de 133.56 minutos.

También llama la atención el vuelo que emplea solo un minuto en el aire (* en la siguiente línea podrá verse tal registro que, como es lógico, se trata de algún fallo en el registro, 'outlier', ya que por ejemplo el tiempo transcurrido de vuelo ya es de 131 minutos, careciendo de toda lógica. O bien se trate del registro de uno de los vuelos detectados por un OVNI, de los que hace apenas unos días ha hablado la NASA. Permítaseme esta breve ocurrencia, ya que estamos hablando de una velocidad media de 42.600 Kilómetros por hora). Po su parte, el máximo tiempo en el aire de un vuelo es de 645 minutos.

El mismo tiempo de retraso de las salidas se traslada al de las llegadas, así se puede ver en sus respectivas medias, las cuales son prácticamente idénticas.

Los datos registrados en los retrasos de llegada son verdaderamente extremos, hay mucha dispersión. Así lo reflejan sus desviaciones típicas, las cuales están incluso por encima de sus respectivas medias. Destacando en los retrasos de llegada un valor en negativo, lo que supone un adelanto en la llegada con respecto a la hora programada de 66 minutos (**).

Por último, cabe resaltar la distancia máxima de 4962 kms y la mínima de 31 kms, también con una desviación típida muy extensa, y con una media de 766.81 kms recorridos por cada vuelo.

In [38]:
# * visualización del resgistro con valor de 1 minuto en el tiempo de vuelo
vuelos_colum_relev[vuelos_colum_relev.AirTime==1]

UniqueCarrier  Month  DayofMonth  DayOfWeek  CRSDepTime  DepTime  \
919536            XE      6          16          1        1120   1143.0   

        CRSArrTime  ArrTime  CRSElapsedTime  ActualElapsedTime  AirTime  \
919536        1325   1354.0           125.0              131.0      1.0   

        DepDelay  ArrDelay Origin Dest  Distance  AvarageSpeed  \
919536      23.0      29.0    TUS  RNO       710       42600.0   

       Delay_WithinTime  TaxiIn_and_TaxiOut  
919536            Delay               130.0

In [11]:
# ** visualización del registro con máximo adelanto de la llegada
vuelos_colum_relev[vuelos_colum_relev.ArrDelay==-66]

UniqueCarrier  Month  DayofMonth  DayOfWeek  CRSDepTime  DepTime  \
50338            OH      1          23          3        1701   1708.0   

       CRSArrTime  ArrTime  CRSElapsedTime  ActualElapsedTime  AirTime  \
50338        1911   1805.0           130.0               57.0     25.0   

       DepDelay  ArrDelay Origin Dest  Distance  
50338       7.0     -66.0    LGA  CMH       478

### b) Datos faltantes por columna


In [13]:
vuelos_colum_relev.isna().sum()


UniqueCarrier          0
Month                  0
DayofMonth             0
DayOfWeek              0
CRSDepTime             0
DepTime                0
CRSArrTime             0
ArrTime              757
CRSElapsedTime        28
ActualElapsedTime    897
AirTime              897
DepDelay               0
ArrDelay             897
Origin                 0
Dest                   0
Distance               0
dtype: int64

In [15]:
print('Si sumamos todos los datos faltantes, hay un total de', vuelos_colum_relev.isna().sum().sum(), 'valores carentes de contenido.')

Si sumamos todos los datos faltantes, hay un total de 3476 valores carentes de contenido.


Como la cantidad de datos nulos son pocos en proporción a la cantidad de datos totales, me inclino más por la aplicación de la función bfill(), mediante el cual se aplica un valor dentro de un rango coherente al tomarlo del valor de la fila siguiente dentro de su respectiva columna. Así, por ejemplo, se evita eliminar toda la fila con la función dropna(), o bien asignar un valor generalizado para todos los datos con el valor nulo (independientemente de la columna a la que pertenezca) con la función fillna().

In [16]:
vuelos_colum_relev=vuelos_colum_relev.bfill()
print('Ahora ya no existen datos faltantes')
vuelos_colum_relev.isna().sum()

Ahora ya no existen datos faltantes


UniqueCarrier        0
Month                0
DayofMonth           0
DayOfWeek            0
CRSDepTime           0
DepTime              0
CRSArrTime           0
ArrTime              0
CRSElapsedTime       0
ActualElapsedTime    0
AirTime              0
DepDelay             0
ArrDelay             0
Origin               0
Dest                 0
Distance             0
dtype: int64

### c) Creación de nuevas columnas

In [23]:
# Velocidad media por vuelo
vuelos_colum_relev['AvarageSpeed']= (vuelos_colum_relev['Distance'] / vuelos_colum_relev['AirTime']*60).round(2)
vuelos_colum_relev

UniqueCarrier  Month  DayofMonth  DayOfWeek  CRSDepTime  DepTime  \
100581             EV      1          28          1        1007   1015.0   
1403335            AS      8           5          2        1845   1852.0   
253588             OO      2          29          5        1710   1918.0   
220528             XE      2          10          7        1351   1401.0   
1508200            AA      9           9          2        1740   1749.0   
...               ...    ...         ...        ...         ...      ...   
437101             OO      3           5          3        1015   1101.0   
1286229            WN      8          22          5        2015   2027.0   
1103904            WN      7          24          4        2115   2148.0   
1614296            AS     10          10          5        1130   1202.0   
191828             WN      2           8          5        1355   1411.0   

         CRSArrTime  ArrTime  CRSElapsedTime  ActualElapsedTime  AirTime  \
100581         1148   1142.0           161.0              147.0    119.0   
1403335        2145   2147.0           180.0              175.0    162.0   
253588         1851   2048.0           101.0               90.0     73.0   
220528         1511   1512.0            80.0               71.0     49.0   
1508200        1905   1854.0           145.0              125.0    103.0   
...             ...      ...             ...                ...      ...   
437101         1221   1254.0           126.0              113.0     97.0   
1286229        2230   2229.0           135.0              122.0    109.0   
1103904        2245   2316.0            90.0               88.0     73.0   
1614296        1554   1627.0           204.0              205.0    175.0   
191828         1520   1546.0            85.0               95.0     55.0   

         DepDelay  ArrDelay Origin Dest  Distance  AvarageSpeed  \
100581        8.0      -6.0    ATL  AUS       813        409.92   
1403335       7.0       2.0    TUS  SEA      1216        450.37   
253588      128.0     117.0    ORD  FSD       462        379.73   
220528       10.0       1.0    LAX  SJC       308        377.14   
1508200       9.0     -11.0    ATL  DFW       732        426.41   
...           ...       ...    ...  ...       ...           ...   
437101       46.0      33.0    DEN  MSO       679        420.00   
1286229      12.0      -1.0    TPA  CMH       829        456.33   
1103904      33.0      31.0    BNA  MCI       491        403.56   
1614296      32.0      33.0    ANC  SEA      1449        496.80   
191828       16.0      26.0    LAX  SFO       337        367.64   

        Delay_WithinTime  
100581        WithinTime  
1403335            Delay  
253588             Delay  
220528             Delay  
1508200       WithinTime  
...                  ...  
437101             Delay  
1286229       WithinTime  
1103904            Delay  
1614296            Delay  
191828             Delay  

[200000 rows x 18 columns]

In [21]:
# Vuelo con retraso o dentro de hora, entendiendo que la hora de llegada efectiva (ArrTime) es la hora real de llegada
# con o sin retraso

import numpy as np

vuelos_colum_relev['Delay_WithinTime']=np.where(vuelos_colum_relev['ArrTime']>vuelos_colum_relev['CRSArrTime'], 'Delay', 'WithinTime')
vuelos_colum_relev

UniqueCarrier  Month  DayofMonth  DayOfWeek  CRSDepTime  DepTime  \
100581             EV      1          28          1        1007   1015.0   
1403335            AS      8           5          2        1845   1852.0   
253588             OO      2          29          5        1710   1918.0   
220528             XE      2          10          7        1351   1401.0   
1508200            AA      9           9          2        1740   1749.0   
...               ...    ...         ...        ...         ...      ...   
437101             OO      3           5          3        1015   1101.0   
1286229            WN      8          22          5        2015   2027.0   
1103904            WN      7          24          4        2115   2148.0   
1614296            AS     10          10          5        1130   1202.0   
191828             WN      2           8          5        1355   1411.0   

         CRSArrTime  ArrTime  CRSElapsedTime  ActualElapsedTime  AirTime  \
100581         1148   1142.0           161.0              147.0    119.0   
1403335        2145   2147.0           180.0              175.0    162.0   
253588         1851   2048.0           101.0               90.0     73.0   
220528         1511   1512.0            80.0               71.0     49.0   
1508200        1905   1854.0           145.0              125.0    103.0   
...             ...      ...             ...                ...      ...   
437101         1221   1254.0           126.0              113.0     97.0   
1286229        2230   2229.0           135.0              122.0    109.0   
1103904        2245   2316.0            90.0               88.0     73.0   
1614296        1554   1627.0           204.0              205.0    175.0   
191828         1520   1546.0            85.0               95.0     55.0   

         DepDelay  ArrDelay Origin Dest  Distance  AvarageSpeed  \
100581        8.0      -6.0    ATL  AUS       813        409.92   
1403335       7.0       2.0    TUS  SEA      1216        450.37   
253588      128.0     117.0    ORD  FSD       462        379.73   
220528       10.0       1.0    LAX  SJC       308        377.14   
1508200       9.0     -11.0    ATL  DFW       732        426.41   
...           ...       ...    ...  ...       ...           ...   
437101       46.0      33.0    DEN  MSO       679        420.00   
1286229      12.0      -1.0    TPA  CMH       829        456.33   
1103904      33.0      31.0    BNA  MCI       491        403.56   
1614296      32.0      33.0    ANC  SEA      1449        496.80   
191828       16.0      26.0    LAX  SFO       337        367.64   

        Delay_WithinTime  
100581        WithinTime  
1403335            Delay  
253588             Delay  
220528             Delay  
1508200       WithinTime  
...                  ...  
437101             Delay  
1286229       WithinTime  
1103904            Delay  
1614296            Delay  
191828             Delay  

[200000 rows x 18 columns]

In [28]:
# Tiempo de carreteo o movimiento del avión sobre tierra
vuelos_colum_relev['TaxiIn_and_TaxiOut']=(vuelos_colum_relev['ActualElapsedTime'] - vuelos_colum_relev['AirTime']).round(2)
vuelos_colum_relev


UniqueCarrier  Month  DayofMonth  DayOfWeek  CRSDepTime  DepTime  \
100581             EV      1          28          1        1007   1015.0   
1403335            AS      8           5          2        1845   1852.0   
253588             OO      2          29          5        1710   1918.0   
220528             XE      2          10          7        1351   1401.0   
1508200            AA      9           9          2        1740   1749.0   
...               ...    ...         ...        ...         ...      ...   
437101             OO      3           5          3        1015   1101.0   
1286229            WN      8          22          5        2015   2027.0   
1103904            WN      7          24          4        2115   2148.0   
1614296            AS     10          10          5        1130   1202.0   
191828             WN      2           8          5        1355   1411.0   

         CRSArrTime  ArrTime  CRSElapsedTime  ActualElapsedTime  AirTime  \
100581         1148   1142.0           161.0              147.0    119.0   
1403335        2145   2147.0           180.0              175.0    162.0   
253588         1851   2048.0           101.0               90.0     73.0   
220528         1511   1512.0            80.0               71.0     49.0   
1508200        1905   1854.0           145.0              125.0    103.0   
...             ...      ...             ...                ...      ...   
437101         1221   1254.0           126.0              113.0     97.0   
1286229        2230   2229.0           135.0              122.0    109.0   
1103904        2245   2316.0            90.0               88.0     73.0   
1614296        1554   1627.0           204.0              205.0    175.0   
191828         1520   1546.0            85.0               95.0     55.0   

         DepDelay  ArrDelay Origin Dest  Distance  AvarageSpeed  \
100581        8.0      -6.0    ATL  AUS       813        409.92   
1403335       7.0       2.0    TUS  SEA      1216        450.37   
253588      128.0     117.0    ORD  FSD       462        379.73   
220528       10.0       1.0    LAX  SJC       308        377.14   
1508200       9.0     -11.0    ATL  DFW       732        426.41   
...           ...       ...    ...  ...       ...           ...   
437101       46.0      33.0    DEN  MSO       679        420.00   
1286229      12.0      -1.0    TPA  CMH       829        456.33   
1103904      33.0      31.0    BNA  MCI       491        403.56   
1614296      32.0      33.0    ANC  SEA      1449        496.80   
191828       16.0      26.0    LAX  SFO       337        367.64   

        Delay_WithinTime  TaxiIn_and_TaxiOut  
100581        WithinTime                28.0  
1403335            Delay                13.0  
253588             Delay                17.0  
220528             Delay                22.0  
1508200       WithinTime                22.0  
...                  ...                 ...  
437101             Delay                16.0  
1286229       WithinTime                13.0  
1103904            Delay                15.0  
1614296            Delay                30.0  
191828             Delay                40.0  

[200000 rows x 19 columns]

### d) Tabla de aerolíneas con más retrasos acumulados. Los cinco primeros al ser los más destacables por su acumulación.

In [29]:
aerol_mas_retr=vuelos_colum_relev.groupby('UniqueCarrier')['ArrDelay'].sum().reset_index().sort_values('ArrDelay', ascending=False).head()
aerol_mas_retr

UniqueCarrier   ArrDelay
17            WN  1182159.0
1             AA   915032.0
15            UA   709540.0
11            MQ   658677.0
14            OO   625967.0

### e)  Los vuelos más largos. Muestra de los 10 primeros para tener una buena visión de esta variable, pero sin llegar a ser excesiva. 
#### # Matizar que el ejercicio no pide una nueva tabla solo con los valores de la columna de 'Distancia' o alguna más, de lo contrario habría procedido como en el ejercicio anterior. Se entiende que presente los registros de los vuelos más largos.

In [31]:
vuelos_colum_relev.sort_values(by='Distance', ascending=False).head(10)

UniqueCarrier  Month  DayofMonth  DayOfWeek  CRSDepTime  DepTime  \
1066860            CO      6          12          4        1335   1532.0   
556365             CO      3           2          7        1320   1353.0   
1621839            CO     10          15          3        1335   1347.0   
1250289            CO      7          19          6        1335   1503.0   
362407             CO      2          18          1        1320   1334.0   
715197             CO      4          12          6        1335   1348.0   
1257658            CO      7          26          6        1335   1355.0   
1925154            CO     12          22          1        2055   2219.0   
875499             CO      5          15          4        1335   1631.0   
365455             CO      2           1          5        1320   1450.0   

         CRSArrTime  ArrTime  CRSElapsedTime  ActualElapsedTime  AirTime  \
1066860        1810   2026.0           635.0              654.0    618.0   
556365         1920   1926.0           660.0              633.0    604.0   
1621839        1810   1812.0           635.0              625.0    599.0   
1250289        1810   1936.0           635.0              633.0    598.0   
362407         1920   1904.0           660.0              630.0    607.0   
715197         1810   1824.0           635.0              636.0    615.0   
1257658        1810   1914.0           635.0              679.0    622.0   
1925154        1120   1259.0           565.0              580.0    554.0   
875499         1810   2152.0           635.0              681.0    645.0   
365455         1920   2036.0           660.0              646.0    620.0   

         DepDelay  ArrDelay Origin Dest  Distance  AvarageSpeed  \
1066860     117.0     136.0    EWR  HNL      4962        481.75   
556365       33.0       6.0    EWR  HNL      4962        492.91   
1621839      12.0       2.0    EWR  HNL      4962        497.03   
1250289      88.0      86.0    EWR  HNL      4962        497.86   
362407       14.0     -16.0    EWR  HNL      4962        490.48   
715197       13.0      14.0    EWR  HNL      4962        484.10   
1257658      20.0      64.0    EWR  HNL      4962        478.65   
1925154      84.0      99.0    HNL  EWR      4962        537.40   
875499      176.0     222.0    EWR  HNL      4962        461.58   
365455       90.0      76.0    EWR  HNL      4962        480.19   

        Delay_WithinTime  TaxiIn_and_TaxiOut  
1066860            Delay                36.0  
556365             Delay                29.0  
1621839            Delay                26.0  
1250289            Delay                35.0  
362407        WithinTime                23.0  
715197             Delay                21.0  
1257658            Delay                57.0  
1925154            Delay                26.0  
875499             Delay                36.0  
365455             Delay                26.0

### Los 10 vuelos con más retrasos. 
#### Procedimiento idem ejercicio anterior. 

In [32]:
vuelos_colum_relev.sort_values(by='ArrDelay', ascending=False).head(10)

UniqueCarrier  Month  DayofMonth  DayOfWeek  CRSDepTime  DepTime  \
686014             NW      4          10          4        1417    724.0   
1027010            NW      6          16          1        1420   1258.0   
1400101            AA      8          29          5         850    630.0   
682367             NW      4           5          6        1200    900.0   
511445             MQ      3          21          5        1025    626.0   
1884133            NW     12          14          7        1420   1026.0   
676451             MQ      4          11          5        1835   1310.0   
1895175            AA     12          18          4         750    213.0   
142186             9E      1          31          4        1345    810.0   
1400321            AA      8           9          6        1435    829.0   

         CRSArrTime  ArrTime  CRSElapsedTime  ActualElapsedTime  AirTime  \
686014         1605    858.0           168.0              154.0    132.0   
1027010        1929   1753.0           189.0              175.0    159.0   
1400101        1230    956.0           160.0              146.0    124.0   
682367         1715   1404.0           195.0              184.0    165.0   
511445         1225    820.0           120.0              114.0     81.0   
1884133        1929   1519.0           189.0              173.0    140.0   
676451         2015   1453.0            40.0               43.0     28.0   
1895175        1550   1008.0           300.0              295.0    270.0   
142186         1714   1128.0           149.0              138.0    115.0   
1400321        1945   1356.0           190.0              207.0    161.0   

         DepDelay  ArrDelay Origin Dest  Distance  AvarageSpeed  \
686014     2467.0    2453.0    CLT  MSP       930        422.73   
1027010    1358.0    1344.0    LAS  MSP      1300        490.57   
1400101    1300.0    1286.0    SLC  DFW       988        478.06   
682367     1260.0    1249.0    PDX  MSP      1426        518.55   
511445     1201.0    1195.0    CVG  LGA       585        433.33   
1884133    1206.0    1190.0    LAS  MSP      1300        557.14   
676451     1115.0    1118.0    GRB  MQT       134        287.14   
1895175    1103.0    1098.0    SEA  JFK      2421        538.00   
142186     1105.0    1094.0    GTF  MSP       887        462.78   
1400321    1074.0    1091.0    RNO  DFW      1345        501.24   

        Delay_WithinTime  TaxiIn_and_TaxiOut  
686014        WithinTime                22.0  
1027010       WithinTime                16.0  
1400101       WithinTime                22.0  
682367        WithinTime                19.0  
511445        WithinTime                33.0  
1884133       WithinTime                33.0  
676451        WithinTime                15.0  
1895175       WithinTime                25.0  
142186        WithinTime                23.0  
1400321       WithinTime                46.0

### Rutas más largas (seleccionadas las 10 primeras)

In [33]:
rutas_mas_larg = vuelos_colum_relev.drop_duplicates(subset = "Distance").sort_values(by = 'Distance', ascending = False).head(10)
rutas_mas_larg[['Origin', 'Dest', 'Distance']]

Origin Dest  Distance
1519542    EWR  HNL      4962
567559     ATL  HNL      4502
845028     ORD  HNL      4243
66614      KOA  ORD      4213
628029     ORD  OGG      4184
517254     MSP  HNL      3972
1621365    IAH  HNL      3904
528485     DFW  HNL      3784
1385570    OGG  DFW      3711
1347931    ANC  ATL      3417

### f) Nuevos datos que considere relevantes

#### Cantidad de vuelos por mes

In [34]:
vuelos_mes = vuelos['Month'].value_counts().reset_index()
vuelos_mes.columns=['Número del mes', 'Cantidad de vuelos']
vuelos_mes.sort_values('Cantidad de vuelos', ascending=False)

#Los meses de Diciembre, Junio y Marzo son en los que más vuelos se realizan, mientras que en el mes de Septiembre es el que menos.

Número del mes  Cantidad de vuelos
0               12               21167
1                6               20871
2                3               20678
3                2               19876
4                1               19115
5                7               18741
6                8               16676
7                4               15978
8                5               15832
9               11               10841
10              10               10542
11               9                9683

#### Cantidad vuelos por día del mes

In [35]:

vuelos_Dia_Mes=vuelos['DayofMonth'].value_counts().reset_index()
vuelos_Dia_Mes.columns=['Día del Mes', 'Cantidad de vuelos']
vuelos_Dia_Mes.sort_values('Cantidad de vuelos', ascending=False)

#Llama la atención como precisamente los últimos días de cada mes (31, 30 y 29) es cuando se realizan menos vuelos,
#en cambio el día 1 está entre los tres primeros

Día del Mes  Cantidad de vuelos
0            22                7462
1            21                7221
2             1                7039
3             7                6961
4            15                6821
5            18                6773
6            24                6698
7            20                6642
8            27                6632
9            14                6628
10           19                6585
11           11                6575
12            2                6570
13           17                6550
14           10                6544
15            8                6532
16           26                6493
17           23                6476
18            6                6445
19            4                6427
20            9                6423
21           12                6408
22           16                6368
23           13                6325
24           28                6282
25            3                6230
26            5                6115
27           25                6051
28           30                5966
29           29                5776
30           31                3982

#### Cantidad vuelos por día de la semana

In [36]:
vuelos_Dia_Sem=vuelos['DayOfWeek'].value_counts().reset_index()
vuelos_Dia_Sem.columns=['Día de la semana', 'Cantidad de vuelos']
vuelos_Dia_Sem.sort_values('Cantidad de vuelos', ascending=False)

## En cuanto a los vuelos que se efectúan por día de la semana, el primer y último día laborables de la semana son los días
# que más vuelos se ejecutan. Por su parte, el sábado es el que menos.

Día de la semana  Cantidad de vuelos
0                 5               33416
1                 4               29934
2                 1               29778
3                 7               29685
4                 3               27261
5                 2               27025
6                 6               22901

## Ejercicio 3

#### Exportar el nuevo dataframe con las nuevas columnas añadidas a Excel

In [37]:
vuelos_colum_relev.to_excel('Vuelos_análisis.xlsx')